src/modules/_nomodule/PixelManipulation.js

```diff
@@ -42,59 +42,104 @@ module.exports = function PixelManipulation(image, options) {
 
     if (options.preProcess) pixels = options.preProcess(pixels); // Allow for preprocessing
 
+    function extraOperation() {
+      var res;
+      if (options.extraManipulation) res = options.extraManipulation(pixels, generateOutput);
+      // there may be a more efficient means to encode an image object,
+      // but node modules and their documentation are essentially arcane on this point
+      function generateOutput() {
+        var chunks = [];
+        var totalLength = 0;
+
+        var r = savePixels(pixels, options.format, {
+          quality: 100
+        });
+
+        r.on('data', function (chunk) {
+          totalLength += chunk.length;
+          chunks.push(chunk);
+        });
+
+        r.on('end', function () {
+          var data = Buffer.concat(chunks, totalLength).toString('base64');
+          var datauri = 'data:image/' + options.format + ';base64,' + data;
+          if (options.output)
+            options.output(options.image, datauri, options.format);
+          if (options.callback) options.callback();
+        });
+      }
+      if (res) {
+        pixels = res;
+        generateOutput();
+      } else if (!options.extraManipulation) generateOutput();
+    }
+
+    if (!options.changePixel) extraOperation();
+
     if (options.changePixel) {
 
       /* Allows for Flexibility
        if per pixel manipulation is not required */
 
-      for (var x = 0; x < pixels.shape[0]; x++) {
-        for (var y = 0; y < pixels.shape[1]; y++) {
-          let pixel = options.changePixel(
-            pixels.get(x, y, 0),
-            pixels.get(x, y, 1),
-            pixels.get(x, y, 2),
-            pixels.get(x, y, 3),
-            x,
-            y
-          );
-
-          pixels.set(x, y, 0, pixel[0]);
-          pixels.set(x, y, 1, pixel[1]);
-          pixels.set(x, y, 2, pixel[2]);
-          pixels.set(x, y, 3, pixel[3]);
-
-          if (!options.inBrowser && !process.env.TEST && options.ui) pace.op();
+      const imports = {
+        env: {
+          consoleLog: console.log,
+          perform: function (x, y) {
+            let pixel = options.changePixel(
+              pixels.get(x, y, 0),
+              pixels.get(x, y, 1),
+              pixels.get(x, y, 2),
+              pixels.get(x, y, 3),
+              x,
+              y
+            );
+
+            pixels.set(x, y, 0, pixel[0]);
+            pixels.set(x, y, 1, pixel[1]);
+            pixels.set(x, y, 2, pixel[2]);
+            pixels.set(x, y, 3, pixel[3]);
+          }
+        }
+      };
+
+      function perPixelManipulation() {
+        for (var x = 0; x < pixels.shape[0]; x++) {
+          for (var y = 0; y < pixels.shape[1]; y++) {
+            imports.env.perform(x,y);
+          }
         }
       }
+
+      const inBrowser = (options.inBrowser) ? 1 : 0;
+      const test = (process.env.TEST) ? 1 : 0;
+      if (options.useWasm) {
+        if (options.inBrowser) {
+
+          fetch('./manipulation.wasm').then(response =>
```

Comment:
Should we move `manipulation.wasm` to the `dist` folder so that it's clear it has to be distributed with the compiled source?

Reply:
I am not sure on how to do that.  Please help me with that!!

Reply:
Can fetch be used like (assuming i got the number of `../` correct...):

```suggestion
          fetch('../../../../dist/manipulation.wasm').then(response =>
```

Reply:
or wait, is it `fetch('./dist/manipulation.wasm')`? I think that's it.

Reply:
is the issue that we're running it from the root dir in tests, but from the `/examples/` dir in the demo?

Reply:
If so, perhaps we actually need to have a constructor parameter to point at it, like:

```js
sequencer = ImageSequencer({
  wasmPath: '../manipulation.wasm'
})
```
?

Reply:
that can have a default at `./manipulation.wasm` but can be set independently in the demo?